Sources/MongoSwift/BSON/BsonEncoder.swift

```diff
@@ -522,10 +522,12 @@ private class MutableArray: BsonValue {
         try self.array.encode(to: storage, forKey: key)
     }
 
+    init() {}
+
     /// methods required by the BsonValue protocol that we don't actually need/use. MutableArray
     /// is just a BsonValue to simplify usage alongside true BsonValues within the encoder.
-    static func from(iter: inout bson_iter_t) -> BsonValue {
-        fatalError("MutableArray is not meant to be initialized from a bson_iter_t")
+    required init(from iter: DocumentIterator) {
+        fatalError("`MutableArray` is not meant to be initialized from a `DocumentIterator`")
```

Comment:
Does it make sense to format classes/types with backticks in error strings? (in contrast to doc blocks, which actually get rendered).